### PR TITLE
Stream files from disk while uploading them to S3

### PR DESCRIPTION
**Pull request by msrb**
Service with 2Gi of memory is getting OOM killed while uploading 650Mi
file.

### CODE REVIEW

**Comment by msrb** [comment]
This is needed to make https://github.com/fabric8-analytics/fabric8-analytics-worker/pull/471 finally work.

**Comment by msrb** [comment]
Tested, merging.

File: f8a_worker/storages/s3.py [hunk 1]
```diff
@@ -154,13 +154,19 @@ def store_file(self, file_path, object_key):
...
+            return self.store_blob(f, object_key)
+        finally:
+            if f is not None:
+                f.close()
 
     def store_blob(self, blob, object_key):
         """ Store blob on S3
 
-        :param blob: bytes to be stored
+        :param blob: bytes or stream to be stored
```
Comment (jpopelka) [inline f8a_worker/storages/s3.py]: [S3.Object.put() doc](http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Object.put) says it's `bytes or seekable file-like object`, which I like more, but I guess this is OK too.

File: f8a_worker/storages/s3.py [hunk 2]
```diff
@@ -154,13 +154,19 @@ def store_file(self, file_path, object_key):
...
-        with open(file_path, 'rb') as f:
-            return self.store_blob(f.read(), object_key)
+
+        f = None
+        try:
+            f = open(file_path, 'rb')
+            return self.store_blob(f, object_key)
+        finally:
+            if f is not None:
+                f.close()
```
Comment (jpopelka) [inline f8a_worker/storages/s3.py]: Looks OK, but why do you think the `with` statement would not work any more ?

**Pull request merged**